underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java

```diff
@@ -264,14 +270,13 @@ public long getSpace(String path, SpaceType type) throws IOException {
 
   @Override
   public boolean isFile(String path) throws IOException {
-    String strippedPath = stripPrefixIfPresent(path);
-    return mAccount.getContainer(mContainerName).getObject(strippedPath).exists();
+    return exists(path) && !isDirectory(path);
```

Comment:
as per our offline discussion, I will leave it as is for the time being and address this across all object storage UFSes in a separate PR